src/AutoRest.CSharp/Output/Models/Types/ParentDetection.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+using System;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Models.Requests;
+using AutoRest.CSharp.Input;
+using System.Linq;
+
+namespace AutoRest.CSharp.Output.Models.Type.Decorate
+{
+    internal static class ParentDetection
+    {
+        public static string GetParent(OperationGroup operationGroup)
+        {
+            if (operationGroup.IsTenantResource)
+            {
+                return TenantDetection.TenantName;
+            }
+            if (operationGroup.IsExtensionResource)
+            {
+                throw new ArgumentException($"Could not set parent for operations group {operationGroup.Key}. This an extensions resource, please add to readme.md");
```

Comment:
I haven't seen any coverage for all these exception scenarios.

Reply:
We need to Spoof the creation of an OperationGroup object which I will create an ADO item.
https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5755

Reply:
@pakrym, we are going to have a multitude of RP swaggers to exercise. The current priority is to create a mainline scenario so we can have at ONE swagger to generate usable SDK. This will allow parallel development both on breath coverage of RP swagger and API adjustment.